File: app/src/main/java/com/example/thekhaeng/viewstatesavetest/view/java/base/ViewSavedState.java
package com.example.thekhaeng.viewstatesavetest.view.java.base;

import android.os.Parcel;
import android.os.Parcelable;
import android.view.View;

/**
 * Created by「 The Khaeng 」on 01 Aug 2018 :)
 */
public abstract class ViewSavedState extends View.BaseSavedState {

    public Parcelable superState;

    public ViewSavedState(Parcelable superState) {
        super(EMPTY_STATE);
        this.superState = superState;
    }


    public ViewSavedState(Parcel in) {
        super(in);
        this.superState = in.readParcelable(getClass().getClassLoader());
    }

    @Override
    public void writeToParcel(Parcel out, int flags) {
        super.writeToParcel(out, flags);
        out.writeParcelable(this.superState, flags);
    }

}

File: app/src/main/java/com/example/thekhaeng/viewstatesavetest/view/kotlin/base/ViewSavedState.kt
package com.example.thekhaeng.viewstatesavetest.view.kotlin.base

import android.os.Bundle
import android.os.Parcel
import android.os.Parcelable
import android.view.View

/**
 * Created by「 The Khaeng 」on 20 Feb 2018 :)
 */

abstract class ViewSavedState : View.BaseSavedState {
    @get:JvmName("getSuperState_")
    var superState: Parcelable = Bundle()

    internal constructor(superState: Parcelable) : super(EMPTY_STATE) {
        this.superState = superState
    }

    internal constructor(`in`: Parcel) : super(`in`) {
        this.superState = `in`.readParcelable(javaClass.classLoader)
    }

    override
    fun writeToParcel(out: Parcel, flags: Int) = with(out) {
        super.writeToParcel(out, flags)
        out.writeParcelable(superState, flags)
    }

}
